aicsimageio/readers/ome_zarr_reader.py

```diff
@@ -0,0 +1,182 @@
+            self._scenes = tuple(
+                metadata_utils.generate_ome_image_id(i)
+                for i in range(len(OmeZarr.root_attrs["multiscales"]))
+            )
+        return self._scenes
+
+    def _read_delayed(self) -> xr.DataArray:
+        return self._xarr_format(delayed=True)
+
+    def _read_immediate(self) -> xr.DataArray:
+        return self._xarr_format(delayed=False)
+
+    def _xarr_format(self, delayed: bool) -> xr.DataArray:
+
+        OmeZarr = ZarrReader(parse_url(self._path, mode="r")).zarr
+        image_data = OmeZarr.load(str(self.current_scene_index))
+        axes = OmeZarr.root_attrs["multiscales"][self.current_scene_index]["axes"]
+        dims = [sub["name"].upper() for sub in axes]
+
+        if not delayed:
+            image_data = image_data.compute()
+
+        channels = self._get_channel_names_for_scene(image_data.shape, dims)
+
+        coords = self._get_coords(
+            dims,
+            image_data.shape,
+            scene_index=self.current_scene_index,
+            channel_names=channels,
+        )
+
+        return xr.DataArray(
+            image_data,
+            dims=dims,
+            coords=coords,
+            attrs={
+                # TODO: ADD info from "multiscales" "omero"?
+            },
+        )
+
+    @staticmethod
+    def _get_coords(
+        dims: List[str],
+        shape: Tuple[int, ...],
+        scene_index: int,
+        channel_names: Optional[List[str]],
+    ) -> Dict[str, Any]:
+        # Use dims for coord determination
+        coords: Dict[str, Any] = {}
+
+        if channel_names is None:
+            # Get ImageId for channel naming
+            image_id = metadata_utils.generate_ome_image_id(scene_index)
+
+            # Use range for channel indices
+            if DimensionNames.Channel in dims:
+                coords[DimensionNames.Channel] = [
+                    metadata_utils.generate_ome_channel_id(
+                        image_id=image_id, channel_id=i
+                    )
+                    for i in range(shape[dims.index(DimensionNames.Channel)])
+                ]
+        else:
+            coords[DimensionNames.Channel] = channel_names
+
+        return coords
+
+    def _get_channel_names_for_scene(
+        self, image_shape: Tuple[int], dims: List[str]
+    ) -> Optional[List[str]]:
+        # Fast return in None case
+        if self._channel_names is None:
+            return None
+
+        # If channels was provided as a list of lists
+        if isinstance(self._channel_names[0], list):
+            scene_channels = self._channel_names[self.current_scene_index]
+        elif all(isinstance(c, str) for c in self._channel_names):
+            scene_channels = self._channel_names  # type: ignore
+        else:
+            return None
+
+        # If scene channels isn't None and no channel dimension raise error
+        if DimensionNames.Channel not in dims:
```

Comment:
Since this method is always called does this then force all ome zarrs to have a channel dim?

Reply:
with [9c2ccf6](https://github.com/AllenCellModeling/aicsimageio/pull/493/commits/9c2ccf6506f33867fc773355c52a38d6c01b6c0e) this becomes that last possible case. Additionally this is used to build coords, it doesnt directly add the channel dim as far as I know